sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared.py

```diff
@@ -353,10 +353,11 @@ def _aiohttp_body_helper(response):
     enc = enc.lower()
     if enc in ("gzip", "deflate"):
         if response._decompressed_content:
```

Comment:
Should we move this to line #350?

Reply:
so before w check for content encoding, if we see that we've already decompressed the response, we just go ahead and return it?

I think that could be valid, that way we don't have to check the response headers each time